### PR TITLE
*: use kvproto::backup::StorageBackend in create_storage (#6186)

### PR DESCRIPTION
**Pull request by kennytm**
###  What have you changed?

Please explain in detail what the changes are in this PR and why they are needed:

- Cherry-picks #6186 and pingcap/kvproto#510 to release-3.1

###  What is the type of the changes?

Pick one of the following and delete the others:

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

###  Refer to a related PR or issue link (optional)

pingcap/br#88

### CODE REVIEW

File: src/import/sst_importer.rs [hunk 1]
```diff
@@ -88,19 +89,19 @@ impl SSTImporter {
...
-        url: &str,
+        backend: &StorageBackend,
         name: &str,
         rewrite_rule: &RewriteRule,
         speed_limiter: Option<Arc<IOLimiter>>,
     ) -> Result<Option<Range>> {
         debug!("download start";
             "meta" => ?meta,
-            "url" => url,
+            "url" => ?backend,
```
Comment (tennix) [inline src/import/sst_importer.rs]: The `backend` may contain secret data. I'm not sure if it's a good idea to log it.

Comment (kennytm) [reply src/import/sst_importer.rs]: @tennix  how about just log the URL `s3://bucket/prefix/`? (we need to forward-port the patch to master)

Comment (tennix) [reply src/import/sst_importer.rs]: There is no url field, for debug purpose, I think all other parameters need to be logged. For the secret, I think we should print `<hidden-secret>` or whatever meaningful for non-empty secret, and `empty` for empty secret.

Comment (kennytm) [reply src/import/sst_importer.rs]: the struct is generated by protobuf/prost so I don't think we can control its debug output (unless we use a custom formatter)

**Comment by kennytm** [comment]
PTAL @tennix @overvenus

**Comment by overvenus** [comment]
Unfortunately, this PR is blocked by #6181 .  😿

**Pull request merged**